sqlglot/dialects/dialect.py

```diff
@@ -1019,3 +1024,45 @@ def merge_without_target_sql(self: Generator, expression: exp.Merge) -> str:
         )
 
     return self.merge_sql(expression)
+
+
+def parse_json_extract_path(
+    expr_type: t.Type[E],
+    supports_null_if_invalid: bool = False,
+) -> t.Callable[[t.List], E]:
+    def _parse_json_extract_path(args: t.List) -> E:
+        null_if_invalid = None
+
+        segments: t.List[exp.JSONPathPart] = [exp.JSONPathRoot()]
+        for arg in args[1:]:
+            if isinstance(arg, exp.Literal):
+                text = arg.name
+                if text.isnumeric():
```

Comment:
let's factor out the is_int from expression.py is_number and use that here

Reply:
ah maybe we can just do `arg.is_int` here, will refactor it

Reply:
well in some engines it can be an int, others it's '1'

Reply:
Can you elaborate on your initial suggestion here a bit? `is_number` is true for literals with the `is_string` arg set to false and `is_int` relies on that logic. Here we inspect the actual text because of what you said in the last reply; are you suggesting we move this as a new property under `Expression` or something?

Reply:
i'm suggesting you factor out the try catch int check, and call that here on text instead of isnumeric

Reply:
something like this then? https://github.com/tobymao/sqlglot/pull/2883/commits/a52a54ab3dc1fc1bc164050e36561bb1657d1621

Reply:
yea, exactly but put it in helpers